lib/distribute.c

```diff
@@ -240,150 +240,62 @@ static int distribute_list_prefix_unset(struct distribute_ctx *ctx,
 	return 1;
 }
 
-DEFUN (distribute_list,
-       distribute_list_cmd,
-       "distribute-list [prefix] WORD <in|out> [WORD]",
-       "Filter networks in routing updates\n"
-       "Specify a prefix\n"
-       "Access-list name\n"
-       "Filter incoming routing updates\n"
-       "Filter outgoing routing updates\n"
-       "Interface name\n")
+static enum distribute_type distribute_direction(const char *dir, bool v4)
 {
-	int prefix = (argv[1]->type == WORD_TKN) ? 1 : 0;
-	/* Check of distribute list type. */
-	enum distribute_type type = argv[2 + prefix]->arg[0] == 'i'
-					    ? DISTRIBUTE_V4_IN
-					    : DISTRIBUTE_V4_OUT;
-
-	/* Set appropriate function call */
-	void (*distfn)(struct distribute_ctx *, const char *,
-		       enum distribute_type, const char *) =
-		prefix ? &distribute_list_prefix_set : &distribute_list_set;
-	struct distribute_ctx *ctx =
-		(struct distribute_ctx *)listnode_head(dist_ctx_list);
-
-	/* if interface is present, get name */
-	const char *ifname = NULL;
-	if (argv[argc - 1]->type == VARIABLE_TKN)
-		ifname = argv[argc - 1]->arg;
+	if (dir[0] == 'i') {
+		if (v4)
+			return DISTRIBUTE_V4_IN;
+		else
+			return DISTRIBUTE_V6_IN;
+	} else if (dir[0] == 'o') {
+		if (v4)
+			return DISTRIBUTE_V4_OUT;
+		else
+			return DISTRIBUTE_V6_OUT;
+	}
 
-	/* Get interface name corresponding distribute list. */
-	distfn(ctx, ifname, type, argv[1 + prefix]->arg);
+	assert("Expecting in or out only, fix your code");
```

Comment:
This assert is always true and never executes, you mean
`assert(!"Expecting in or out only, fix your code");`

Reply:
fixed